kraken-std/src/kraken/std/cargo/__init__.py

```diff
@@ -405,16 +422,32 @@ def cargo_check_toolchain_version(
 
     project = project or Project.current()
     task = project.task(
-        f"cargoCheckVersion/{minimal_version}", CargoCheckToolchainVersionTask, group=CARGO_BUILD_SUPPORT_GROUP_NAME
+        f"cargoCheckVersion/{minimal_version}",
+        CargoCheckToolchainVersionTask,
+        group=CARGO_BUILD_SUPPORT_GROUP_NAME,
     )
     task.minimal_version = minimal_version
     return task
 
 
-def rustup_target_add(target: str, *, group: str | None = None, project: Project | None = None) -> RustupTargetAddTask:
+def rustup_target_add(
+    target: str, *, group: str | None = None, project: Project | None = None
+) -> RustupTargetAddTask:
     """Creates a task that installs a given target for Cargo"""
 
     project = project or Project.current()
     task = project.task(f"rustupTargetAdd/{target}", RustupTargetAddTask, group=group)
     task.target = target
     return task
+
+
+def cargo_generate_deb_package(*, project: Project | None = None, package_name: str):
+    project = project or Project.current()
+    task = project.do(
+        "cargoGenerateDeb",
+        CargoGenerateDebPackage,
+        package_name=package_name,
+        group="publish",
+    )
+    task.depends_on(":cargoBuildRelease")
```

Comment:
It is a bit annoying because the task may not always be called `cargoBuildRelease`, it can also be `cargoBuildDebug`: https://github.com/kraken-build/kraken-build/blob/3011038a7b8c671c606c7c9f97e6b71771914a1c/kraken-std/src/kraken/std/cargo/__init__.py#L291-L292

Perhaps the cleaner solution here is to have `CargoGenerateDebPackage` accept a list of `package_name`s which can be `list[str] | list[Property[BinaryArtifact]]`.

Then, when calling `cargo_generate_deb_package`, you'd pass it the path to the binaries coming from the build task:

https://github.com/kraken-build/kraken-build/blob/3011038a7b8c671c606c7c9f97e6b71771914a1c/kraken-std/src/kraken/std/cargo/tasks/cargo_build_task.py#L46

Because if a task has a property which is another task's output property, kraken automatically creates a dependency between the two.

Reply:
I think this should be fine, as we only want to package release binaries for now. It could definitely be improved in the longer run.

Reply:
Still, we cannot leave it as is because sometimes the build task will not exist: for example, it might be `cargoBuildRelease` on develop, and `cargoBuildDebug` on a feature branch, in which case this will fail.

If we don't want to automatically generate the dependency via `Property`s, then we should defer to the caller (who will know better) the responsibility of adding the dependency

Reply:
Unfortunately, it is not _the task_ that would fail, but the generation of the task graph. Meaning that you would not be able to run _any_ kraken task, as building the dependency graph would fail with an error (something like `No task matched the selector ':**:cargoBuildRelease'`)

Reply:
Just checked and I think I can safely remove the dependency on any `cargoBuild*`. `cargo deb` seems to build an optimised binary even if `cargo build` wasn't run first, so it will work in any situation.

And it being part of the `publish` group should limit repeating work.